src/Modules/Microsoft.Azure.Functions.PowerShellWorker/Microsoft.Azure.Functions.PowerShellWorker.psm1

```diff
@@ -4,16 +4,26 @@
 #
 
 using namespace System.Management.Automation
+using namespace System.Management.Automation.Runspaces
+using namespace Microsoft.Azure.Functions.PowerShellWorker
 
 # This holds the current state of the output bindings.
 $script:_OutputBindings = @{}
+$script:_FuncMetadataType = "FunctionMetadata" -as [type]
+$script:_RunningInPSWorker = $null -ne $script:_FuncMetadataType
 # These variables hold the ScriptBlock and CmdletInfo objects for constructing a SteppablePipeline of 'Out-String | Write-Information'.
 $script:outStringCmd = $ExecutionContext.InvokeCommand.GetCommand("Microsoft.PowerShell.Utility\Out-String", [CommandTypes]::Cmdlet)
 $script:writeInfoCmd = $ExecutionContext.InvokeCommand.GetCommand("Microsoft.PowerShell.Utility\Write-Information", [CommandTypes]::Cmdlet)
 $script:tracingSb = { & $script:outStringCmd -Stream | & $script:writeInfoCmd -Tags "__PipelineObject__" }
 # This loads the resource strings.
 Import-LocalizedData LocalizedData -FileName PowerShellWorker.Resource.psd1
 
+# Enum that defines different behaviors when collecting output data
+enum DataCollectingBehavior {
+    Scalar
```

Comment:
Do you consider a single object (i.e. hashtable) as a `Scalar`? I feel like `Singleton` might be better here. Not blocking on this though.

Reply:
Yes, I also believe `Singleton` is better. Will change.